src/validation-dispatcher/record-dispatcher.ts

```diff
@@ -0,0 +1,18 @@
+import { ValidationResult, RecordValidationFunction } from '../model';
+import { areAllElementsInArrayDefined } from '../helper';
+
+export const fireRecordValidations = (
+  values: any,
+  validations: RecordValidationFunction[]
+): Promise<ValidationResult>[] => {
+  const validationResultsPromises: Promise<ValidationResult>[] = [];
+
+  if (areAllElementsInArrayDefined(validations)) {
+    validations.forEach(validationFn => {
```

Comment:
If you want, you could use `map` method instead of `forEach + `push`:

```
validationResultsPromises = validations.map(validationFn =>
      validationFn(values)
    );
```

Reply:
good catch, updating it